src/core/spmatrix.c

```diff
@@ -581,14 +581,18 @@ int igraph_spmatrix_add_rows(igraph_spmatrix_t *m, long int n) {
 
 /**
  * \function igraph_spmatrix_clear_row
- * \brief Clears a row in the matrix (sets all of its elements to zero)
+ * \brief Clears a row in the matrix (sets all of its elements to zero).
  * \param m The matrix.
  * \param row The index of the row to be cleared.
+ * \return Error code. The current implementation always succeeds.
```

Comment:
THis is not true, you have just added a code path that returns `IGRAPH_EINVAL` :)